### DISCUSSION TITLE
Using "HiHealth" in the future.

### DISCUSSION THREAD

**Discussion by mbraeutig** [CTHRU/Huawei-TCX-Converter#26] [Ideas]
Hi,

please have a look at:
https://forums.developer.huawei.com/forumPortal/en/topicview?tid=0201339947472930085&fid=0101187876626530001
and here:
https://developer.huawei.com/consumer/en/hms/huaweihealth/

Maybe you can use this in the future or ?

I do not understand all,
but i think, there is a little bit hope, so we need not every time request our data.

greet and thanks
Michael

**Comment by CTHRU** [comment]
Thanks for the heads up, and that is indeed good news.
I already looked into that API, but it is (was) not possible to get the detailed data via the existing API (except for some summary information of the activity).

**Comment by kelvindavies11** [comment]
Hi team,
Firstly thank you for the service you are providing - Top Marks!!!

I'm still reading through the supporting documentation, but Huawei seem to be making progress with providing access to activity data :-)

Activity types are now available here - https://developer.huawei.com/consumer/en/doc/activity-types-0000001050071667-V5

I'm struggling to find specifics about available GPS activity data, but the following scope page suggests you get use some calls to get activity data (https://www.huawei.com/healthkit/activityrecord.read) and also location data. - https://developer.huawei.com/consumer/en/doc/scope-list-0000001055419280-V5


Example on how to read activities: https://developer.huawei.com/consumer/en/doc/development/HMSCore-Guides-V5/read-sports-recording-scene-0000001050782026-V5

Key Classes:
ActivityRecordResult - https://developer.huawei.com/consumer/en/doc/development/HMSCore-References-V5/activityrecordresult-0000001050093261-V5

ActivityRecord - https://developer.huawei.com/consumer/en/doc/development/HMSCore-References-V5/activityrecord_class-0000001050161730-V5

ActivitySummary - https://developer.huawei.com/consumer/en/doc/development/HMSCore-References-V5/activity-summary-0000001055215648-V5

SampleSet - https://developer.huawei.com/consumer/en/doc/development/HMSCore-References-V5/sampleset_class-0000001050164521-V5

SamplePoint - https://developer.huawei.com/consumer/en/doc/development/HMSCore-References-V5/samplepoint_class-0000001050164515-V5

SportDataTypes - https://developer.huawei.com/consumer/en/doc/development/HMSCore-References-V5/sportdatatypes-0000001050090882-V5

**Reply by CrooZER** [reply]
Hello guys. As I understood Huawei completed or partially completed API support for their Health services.
I registered as developer in their console, created simple node.js app. I've added oAuth and successfully logged. Also I've got Bearer token for requests. But I can't get info yet, because their datacollectors stuff confused me. Response said I don't have enough scopes to get this info. So I keep trying and if I get activities info I will let you know. 
Also I noticed Strava has good support(incl. JS, Python, simple HTTP...). 
https://developers.strava.com/docs/reference/
So my question is - Will you try this way of getting info, and create some app\site with Heawei and Strava auth to sync them, or you aren't interested in this? You're a great guys, and I hope you can help millions Huawei customers finally have this integration w\o pain. Happy holidays!

**Reply by CTHRU** [reply]
To answer your question, yes, I'm planning to implement syncing, unfortunately, due to personal reasons, I can't spend much time on it.
As for the Hitrava Web version, I have a working alpha that syncs with Strava using their API, however I still have some work on the back-end side (we wouldn't want the Strava API key wondering around, do we).
I'm definitely interested to do the Huawei side too, but I will concentrate first on getting that Strava version out. Last time I checked, Huawei was still adding functionality to the API (and as you have also noticed, it will take some time to figure out which API to use and how to use it properly).

**Comment by kelvindavies11** [comment]
Hi,
Happy to help, but pleasenote nodejs isn't my tech stack. Is the developer program safe? I'm a little concerned they wanta copy of driving license and bank details.

Kelvin

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
